src/ipfsd-in-proc.js

```diff
@@ -73,29 +75,23 @@ class InProc extends EventEmitter {
       libp2p: this.opts.libp2p,
       config: this.opts.config
     })
-
-    // TODO: should this be wrapped in a process.nextTick(), for context:
-    // https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/#why-use-process-nexttick
-    this.exec.once('error', err => this.emit('error', err))
-    this.exec.once('ready', () => this.emit('ready'))
+    this.exec.once('error', cb)
+    this.exec.once('ready', () => cb(null, this))
   }
 
-  /**
-   * Get the address of connected IPFS API.
-   *
-   * @member {Multiaddr}
-   */
-  get apiAddr () {
-    return this._apiAddr
+  setApi (addr) {
+    this.apiAddr = multiaddr(addr)
+    this.api = (this.opts.IpfsApi || IpfsApi)(addr)
+    // TODO find out why we set this
+    this.api.apiHost = this.apiAddr.nodeAddress().address
+    this.api.apiPort = this.apiAddr.nodeAddress().port
   }
 
-  /**
-   * Get the address of connected IPFS HTTP Gateway.
-   *
-   * @member {Multiaddr}
-   */
-  get gatewayAddr () {
-    return this._gatewayAddr
+  setGateway (addr) {
+    this.gatewayAddr = multiaddr(addr)
+    // TODO find out why we set this
```

Comment:
did you have any conclusion on it? The `in-proc` node should not need this afaik

Reply:
i'll get back to this in another PR. i have added this to my todo and will about an issue about it.